dbms/src/Columns/ColumnConst.cpp

```diff
@@ -55,17 +55,19 @@ ColumnPtr ColumnConst::filter(const Filter & filt, ssize_t /*result_size_hint*/)
     return ColumnConst::create(data, countBytesInFilter(filt));
 }
 
-ColumnPtr ColumnConst::replicate(const Offsets & offsets) const
+ColumnPtr ColumnConst::replicateRange(size_t /*start_row*/, size_t end_row, const IColumn::Offsets & offsets) const
 {
     if (s != offsets.size())
         throw Exception(
             fmt::format("Size of offsets ({}) doesn't match size of column ({})", offsets.size(), s),
             ErrorCodes::SIZES_OF_COLUMNS_DOESNT_MATCH);
 
-    size_t replicated_size = 0 == s ? 0 : offsets.back();
+    assert(end_row <= s);
+    size_t replicated_size = 0 == s ? 0 : (offsets[end_row - 1]);
```

Comment:
So, the replicated_size of `ColumnConst::replicateRange(0, s, offsets)` and `ColumnConst::replicateRange(s - 1, s, offsets)` is the same?

Reply:
If I call `ColumnConst::replicateRange(0, 10, offsets)` and `ColumnConst::replicateRange(10, s, offsets)`, the total row count will be `offsets[9] + offsets[s - 1]`, and it is not equal to the row count of `ColumnConst::replicateRange(0, s, offsets) = offsets[s - 1]`

Reply:
The index of offsets that does not participate in the replication is always zero.

Reply:
The value of offsets that does not participate in the replication can be ignore, it is useless.